File: packages/domain/dto/status.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

const (
	StatusStateUpToDated      = "UP_TO_DATED"
	StatusStateNeedsUpdate    = "NEEDS_UPDATE"
	StatusStateScanningFailed = "SCANNING_FAILED"
)

type Status struct {
	State string `json:"state" binding:"required"`
	Info  string `json:"info"`
}

type StatusOnSystemResponse struct {
	Status

	ID        uuid.UUID `json:"id" binding:"required"`
	CreatedAt time.Time `json:"createdAt" binding:"required"`
	UpdatedAt time.Time `json:"updatedAt" binding:"required"`
}

type StatusOnWebsitesResponse struct {
	Status

	UpdatedAt time.Time `json:"updatedAt" binding:"required"`
}

File: packages/dashboard/src/types/dtos/status.ts
enum StatusState {
  UpToDated = "UP_TO_DATED",
  NeedsUpdate = "NEEDS_UPDATE",
  ScanningFailed = "SCANNING_FAILED",
}

type StatusOnWebsitesResponse = {
  state: StatusState;
  info: string;
  updatedAt: string;
};

export { StatusState };
export type { StatusOnWebsitesResponse };
